### ISSUE TITLE
Tutorial Enhancement on Customizing Models

### ISSUE THREAD

**Issue by shakib221b** [facebookresearch/detectron2#1847]
## 🚀 Feature
Expand Detectron2 tutorial on writing/customizing models: https://detectron2.readthedocs.io/tutorials/write-models.html

## Motivation & Examples

For my work, I have modified Detectron2's Mask-RCNN with Efficientnet backbone, pretrained on COCO dataset and finetuned on a custom dataset [instance segmentation task]. It has been a very good learning experience for me. From my experience, I felt a more detailed tutorial on how to customize/write parts of a model may be helpful for beginner-intermediate coders like me. The current example shows a toy backbone creation but how to modify it to build a working Faster-RCNN or Mask-RCNN model may not be straightforward for those who are new to this library. 

If it is relevant or useful to the commmunity, I'd be happy to share a colab notebook explaining customizing parts of a model in details.

**Comment by alexriedel1** [comment]
Yeah please share your experiences and findings!

**Comment by osain-az** [comment]
That will be useful or helpful in fast-tracking  learning

**Comment by ppwwyyxx** [comment]
We made some updates in https://detectron2.readthedocs.io/tutorials/write-models.html with more details.
I'm not sure if there is anything more that should be added. If you have any specific suggestions feel free to comment.

**Comment by nadyadtm** [comment]
I think we need the .ipynb implementation, so we can understand the step more. And we need more example case like how to change optimizer, change backbone, etc
